src/v/cluster/rm_stm.cc

```diff
@@ -247,6 +247,7 @@ rm_stm::rm_stm(
 }
 
 bool rm_stm::check_tx_permitted() {
+    // TODO support compaction
```

Comment:
unrelated change

Reply:
seems related to me. Did you want this pulled out to a separate commit?

Reply:
I'll pull the comments out into a separate patch now, resolving.